### PR TITLE
add new rules for backend

### PR DESCRIPTION
**Pull request by Fun795**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,7 +25,7 @@
   },
   "homepage": "https://github.com/Byndyusoft/eslint-config#readme",
   "devDependencies": {
-    "eslint": "^8.13.0",
+    "eslint": "^8.22.0",
```
Comment (KillWolfVlad) [inline package.json]: update ESLint in peerDependencies too

File: typescript-style-backend.js [hunk 2]
```diff
@@ -4,6 +4,19 @@ module.exports = {
             files: ['*.ts'],
             plugins: ['simple-import-sort'],
             rules: {
+                camelcase: [
```
Comment (KillWolfVlad) [inline typescript-style-backend.js]: This case doesn't work:

```typescript
export class A {
  public UPPERCASE(): void {
    console.log("1");
  }
}

export class B extends A {
  public constructor() {
    super();
    this.UPPERCASE();
  }
}
```

and

```typescript
export class A {
  public UPPERCASEFIELD = 0;
}

export class B extends A {
  public constructor() {
    super();
    console.log(this.UPPERCASEFIELD);
  }
}
```

File: typescript-style-backend.js [hunk 3]
```diff
@@ -4,6 +4,19 @@ module.exports = {
...
+                    'error',
+                    {
+                        properties: 'always'
+                    }
+                ],
+                'no-underscore-dangle': [
```
Comment (KillWolfVlad) [inline typescript-style-backend.js]: The error occurs during the call, not during the declaration.

image

**Review by KillWolfVlad — CHANGES_REQUESTED** [review]

**Comment by KillWolfVlad** [comment]
I think better use https://typescript-eslint.io/rules/naming-convention rule

```patch
diff --git a/typescript-style-backend.js b/typescript-style-backend.js
index 1598c18..a657987 100644
--- a/typescript-style-backend.js
+++ b/typescript-style-backend.js
@@ -108,6 +108,12 @@ module.exports = {
                     {
                         selector: 'enumMember',
                         format: ['camelCase']
+                    },
+                    {
+                        selector: ['classProperty', 'classMethod', 'parameterProperty'],
+                        format: ['camelCase'],
+                        leadingUnderscore: 'forbid',
+                        trailingUnderscore: 'forbid'
                     }
                 ],
                 // @typescript-eslint/eslint-plugin Extension Rules
```

File: package.json (hunk 1 above)
Comment (Fun795) [reply package.json]: updated

File: typescript-style-backend.js (hunk 2 above)
Comment (Fun795) [reply typescript-style-backend.js]: add rules for class method

**Review by KillWolfVlad — CHANGES_REQUESTED** [review]
> feat: add new rules

Please, use more specific message

**Review by KillWolfVlad — APPROVED** [review]

**Pull request merged**